consensus/src/ledger/mod.rs

```diff
@@ -77,10 +78,18 @@ pub trait Ledger: Send + Sync {
 
     /// checks if a ledgers state is consistent
     fn validate_ledger(&self) -> bool;
+
+    fn requires_async_task(&self, new_commitments_len: usize, new_serial_numbers_len: usize) -> bool;
 }
 
 pub struct DynLedger(pub Box<dyn Ledger>);
 
+impl Default for DynLedger {
+    fn default() -> Self {
+        DynLedger(Box::new(dummy::DummyLedger))
```

Comment:
why do we want a panicky default here?

Reply:
that's just so that `mem::take` can work; but fair, I can just use `mem::replace` to avoid accidental use of `DummyLedger` :+1: